tests/test_errors.py

```diff
@@ -52,3 +53,27 @@ def test_errors():
 
     with pytest.raises(audformat.errors.BadKeyError):
         db['bad']
+
+    with pytest.raises(audformat.errors.BadKeyError):
+        db.media['bad']
+
+    with pytest.raises(audformat.errors.BadKeyError):
+        db.misc_tables['bad']
+
+    with pytest.raises(audformat.errors.BadKeyError):
+        db.raters['bad']
+
+    with pytest.raises(audformat.errors.BadKeyError):
+        db.schemes['bad']
+
+    with pytest.raises(audformat.errors.BadKeyError):
+        db.splits['bad']
+
+    with pytest.raises(audformat.errors.BadKeyError):
+        db.tables['bad']
+
+    with pytest.raises(ValueError):
+        db['misc'] = audformat.Table()
+
+    with pytest.raises(ValueError):
+        db['files'] = audformat.MiscTable(pd.Index([]))
```

Comment:
Why does this raise a `ValueError`?

There is also no `Raises` section in the docstring of `MiscTable`

Reply:
I replaced `ValueError` with a custom class `TableExistsError` like we do for other errors and added the missing raised section in `Database.__setitem()`. I also added a comment to the test and updated the description of the PR accordingly.

Reply:
What happens if you try to set a filewise table with the same ID as another already existing filewise table?

Reply:
Then the old table is simply replaced. Same when you replace a filewise with a segmented table or vice versa. Or when you replace a misc table with a misc table.

Reply:
As we share the same table IDs, I think it would make more sense if we also allow replacing a filewise with a misc table, etc.

Reply:
Mhh ok, but since we plan to reference `MiscTable`s in schemes it can then happen that we lose the labels.

Reply:
That is a valid point, but to a user it feels strange that they are more protected than a filewise table, which I can replace by a segmented table.

Is there maybe a way to check if such a table is used in a scheme, and only then raise the error (when we implement that feature)

Reply:
> The other option we have is to generally forbid that a user can replace an existing table.

Which will also not really help as the user can use `drop_tables()` or `del` to first remove the table and then assign a different one under the same name.

I would propose that for now we allow for replacing the tables and cover raising an error when we tackle `MiscTable` as a scheme.

Reply:
> Which will also not really help as the user can use drop_tables() or del to first remove the table and then assign a different one under the same name.

This currently works only for `Table` objects and not `MiscTable`. And it still has the advantage that the user has to explicitly drop the tables first. Currently, it can happen that a user accidentally uses the same table ID twice.

Reply:
On the other hand, you are right the user can always use `del`, also with schemes etc. So maybe we simply assume that a user takes care of not overwriting it by accident. Will change it accordingly.

Reply:
Oh wait, there is of course one problem: the user can now do: `db.tables['table'] = Table()` and `db.misc_tables['table'] = MiscTable()`.

Reply:
> This is how the current implementation works, right?

yes

Reply:
Good, then I would now also be in favor of using this and will resolve this.